app/grants/management/commands/round8_payouts.py

```diff
@@ -0,0 +1,433 @@
+SCALE = Decimal(1e18) # scale factor for converting Dai units
+WAIT_TIME_BETWEEN_TXS = 15 # seconds
+
+class Command(BaseCommand):
+
+    help = 'Sets or verifies the match payouts in the Grants Round 8 Payout contract.'
+
+    def add_arguments(self, parser):
+        # Required arguments
+        parser.add_argument('what',
+            default='verify',
+            type=str,
+            help="what do we do? (finalize, payout_test, prepare_final_payout, verify, set_payouts_test, set_payouts)"
+        )
+        parser.add_argument('network',
+            default='rinkeby',
+            type=str,
+            help="Must be either 'mainnet' or 'rinkeby'"
+        )
+
+        # Required if what != verify
+        parser.add_argument('--clr_pks',
+            default=None,
+            type=str,
+            help="what CLR PKs should we payout? (eg 1,2,3,4)"
+        )
+        parser.add_argument('--clr_round',
+            default=None,
+            type=int,
+            help="what CLR round number is this? eg 7"
+        )
+        
+
+    def handle(self, *args, **options):
+
+        # Parse inputs
+        what = options['what']
+        network = options['network']
+
+        valid_whats = ['finalize', 'payout_test', 'prepare_final_payout', 'verify', 'set_payouts_test', 'set_payouts']
+        if what not in valid_whats:
+            raise Exception(f"Invalid value {what} for 'what' arg")
+        if network not in ['rinkeby', 'mainnet']:
+            raise Exception(f"Invalid value {network} for 'network' arg")
+
+        # Define parameters that vary by network
+        from_block = 11466409 if network == 'mainnet' else 7731622 # block contract was deployed at
+        dai_address = '0x6B175474E89094C44Da98b954EedeAC495271d0F' if network == 'mainnet' else '0x2e055eEe18284513B993dB7568A592679aB13188'
+        expected_total_dai_amount = 100_000 if network == 'mainnet' else 5000 # in dollars, not wei, e.g. 500 = 500e18
+        
+        # Get contract instances
+        PROVIDER = "wss://" + network + ".infura.io/ws/v3/" + settings.INFURA_V3_PROJECT_ID
+        w3 = Web3(Web3.WebsocketProvider(PROVIDER))
+
+        match_payouts_address = '0xAf32BDf2e2720f6C6a2Fce8B50Ed66fd2b46d478' # same on mainnet and Rinkeby
```

Comment:
i think this could go onto the GrantCLR object.  it might be different per round (we dont want to throw all the DAI in one contract, as that centralises security risks)

Reply:
Sorry I'm not sure I follow the suggestion here?

This is the MatchPayouts contract address that will hold all DAI, and that grant owners will withdraw from. I named this file to make it clear that this script is only for round 8 (since we discussed potentially improving things further going forward).

Reply:
there are multiple subrounds (see the grantclr object in the admin) in the larger round 8 round.  i guess its a design decision whether we want to have all 300k of the ethereum round + the 200k of the apollo round all in one big smart contract.  seems like more risk to have it all in 1 smart contract to me, but i havent thought deeply about.